extensions/http/src/main/java/org/mule/extension/http/api/listener/HttpListenerProvider.java

```diff
@@ -0,0 +1,175 @@
+     * The value of this attribute is only used when persistent connections are enabled.
+     */
+    @Parameter
+    @Optional(defaultValue = "30000")
+    @Expression(NOT_SUPPORTED)
+    private Integer connectionIdleTimeout;
+
+    /**
+     * If false, each connection will be closed after the first request is completed.
+     */
+    @Parameter
+    @Optional(defaultValue = "true")
+    @Expression(NOT_SUPPORTED)
+    private Boolean usePersistentConnections;
+
+    @Inject
+    private HttpListenerConnectionManager connectionManager;
+    @Inject
+    private MuleContext muleContext;
+
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        if (port == null)
+        {
+            port = protocol.getDefaultPort();
+        }
+
+        if (protocol.equals(HTTP) && tlsContext != null)
+        {
+            throw new InitialisationException(CoreMessages.createStaticMessage("TlsContext cannot be configured with protocol HTTP. " +
+                                                                               "If you defined a tls:context element in your listener-config then you must set protocol=\"HTTPS\""), this);
+        }
+        if (protocol.equals(HTTPS) && tlsContext == null)
+        {
+            throw new InitialisationException(CoreMessages.createStaticMessage("Configured protocol is HTTPS but there's no TlsContext configured"), this);
+        }
+        if (tlsContext != null && !tlsContext.isKeyStoreConfigured())
+        {
+            throw new InitialisationException(CoreMessages.createStaticMessage("KeyStore must be configured for server side SSL"), this);
+        }
+
+        if (tlsContext != null)
+        {
+            LifecycleUtils.initialiseIfNeeded(tlsContext);
+        }
+
+        verifyConnectionsParameters();
+    }
+
+    @Override
+    public Server connect(HttpListenerConfig httpListenerConfig) throws ConnectionException
+    {
+        HttpServerConfiguration serverConfiguration = new HttpServerConfiguration.Builder()
+                .setHost(host)
+                .setPort(port)
+                .setTlsContextFactory(tlsContext)
+                .setUsePersistentConnections(usePersistentConnections)
+                .setConnectionIdleTimeout(connectionIdleTimeout)
+                .setOwnerName(httpListenerConfig.getConfigName())
+                .build();
+        Server server = connectionManager.create(serverConfiguration);
+
+        try
+        {
+            server.start();
+        }
+        catch (IOException e)
+        {
+            throw new ConnectionException("Could not start server", e);
+        }
+
+        return server;
+    }
+
+    @Override
+    public void disconnect(Server server)
+    {
+        server.stop();
+    }
+
+    @Override
+    public ConnectionValidationResult validate(Server server)
+    {
+        return ConnectionValidationResult.success();
```

Comment:
I can check whether it's stopped or I could try sending a request to it's server address, would the usual scenario for most connections be a full check?